Updated Excel Data, logintest & Registrationtest

llD3vi1ll opened the pull request:
1) Updated Excel user data
2) Utilized the excel data to login test
3) Utilized the excel data to Registration test.

llD3vi1ll committed:
Updated Excel Data, logintest & Registrationtest

1) Updated Excel user data
2) Utilized the excel data to login test
3) Utilized the excel data to Registration test.

RAKESH13595 commented on src/main/java/pages/LoginPage.java:
```diff
@@ -4,19 +4,24 @@
 		actionDriver.performAction(ActionType.ENTER_TEXT, ElementLocators.EMAIL_TEXT, username);
 		actionDriver.performAction(ActionType.ENTER_TEXT, ElementLocators.PASSWORD_TEXT, password);
 	}
+	public void populateDashBoardPage() {
```
Can we rename the method name to the action being performed.
Ex: performLogoutAction

llD3vi1ll replied:
Done have changed it to logout action.

RAKESH13595 commented on src/test/java/tests/CreateAccountTest.java:
```diff
@@ -36,201 +38,220 @@ public void setupPage(Method method) {
+	@Test(dataProvider = "excelData", dataProviderClass = ExcelDataProvider.class)
+	public void testSuccessfulAccountCreation(Map<String, String> data) throws Exception {
+		createAccountPage.populateCreateAccountFields(data.get("First Name"), data.get("Lirst Name"), data.get("Email"),
+				data.get("Passward"), data.get("Confirm Passward"));
```
instead of doing this for every test case, can we pass the data variable to populateCreateAccountFields and extract the field data inside the populateCreateAccountFields.

RAKESH13595 commented on src/test/java/tests/LoginTest.java:
```diff
@@ -42,287 +40,324 @@ public void testValidLogin(Map<String, String> data) {
-		loginPage.populateLoginFields("JOHN.DOE@EXAMPLE.COM", "Password123!");
+	@Test(dataProvider = "excelData", dataProviderClass = ExcelDataProvider.class)
+	public void testCaseSensitivityEmail(Map<String, String> data) {
+		loginPage.populateLoginFields(data.get("Email"), data.get("Password"));
```
Similar to above CreateAccount test data. Pass the data object to populateLoginFields and read the data inside the method.

RAKESH13595 commented on src/test/java/tests/LoginTest.java:
```diff
@@ -42,287 +40,324 @@ public void testValidLogin(Map<String, String> data) {
+				"Hello, " + expectedDashBoardMessage));
+		loginPage.populateDashBoardPage();
+		Assert.assertTrue(getActionDriver().verifyAttribute(ElementLocators.EMAIL_TEXT, "value", expectedEmailText),
+				"Expected Email field to be value='john.doe@example.com' after logout");
```
Is this message correct? Also do we require this message? Is it used in report?

llD3vi1ll replied:
removed the messge. its only required when test case failed.

RAKESH13595 commented on src/main/java/base/BaseTest.java:
```diff
@@ -22,6 +22,7 @@
 
 import actiondriver.ActionDriver;
+import utils.ApiRequestHandler;
 import utils.ConfigReader;
```
Is this needed?

llD3vi1ll replied:
Yes, when we register new user, and test case fails due to assert value not matching the next line which is deleting user fails to delete.
So been using after suite to delete user, if test case fails. That's why it's required.

![image](https://github.com/user-attachments/assets/2b886b70-0f00-4a1e-914c-41b425805fa6)

RAKESH13595 replied:
Can you try softAssert for these test cases? @llD3vi1ll

llD3vi1ll committed:
Optimized testcase

Optimized Login and register test case.

llD3vi1ll commented:
@RAKESH13595 can you check the updates code

llD3vi1ll committed:
Test Data update.

Test Data updated

llD3vi1ll committed:
Updated TestData and changes in testcase

Added registration Test Data and changes in test cases

llD3vi1ll committed:
Updates on all Test's

1) added more test case to logout test.
2) error correction on create account test
3) moved some test case from login to forgot test.

llD3vi1ll commented:
Final updates. Check and advise if any changes to be made.

RAKESH13595 reviewed (approved):
LGTM
